### PR TITLE
chore(authz): increase coverage

### PR DESCRIPTION
**Pull request by savonarola**
Also added some small fixes.

### CODE REVIEW

File: apps/emqx_authz/src/emqx_authz_schema.erl [hunk 1]
```diff
@@ -153,11 +152,6 @@ mongo_common_fields() ->
                  default => true}}
     ].
 
-validations() ->
```
Comment (savonarola) [inline apps/emqx_authz/src/emqx_authz_schema.erl:156]: `emqx_authz_schema` is never used directly, so these validations are never called.

File: apps/emqx_authz/src/emqx_authz_api_sources.erl [hunk 2]
```diff
@@ -445,7 +445,8 @@ read_certs(#{<<"ssl">> := SSL} = Source) ->
     end;
 read_certs(Source) -> Source.
 
-maybe_write_certs(#{<<"ssl">> := #{<<"enable">> := true} = SSL} = Source) ->
+maybe_write_certs(#{<<"ssl">> := #{<<"enable">> := True} = SSL} = Source)
+  when (True =:= true) or (True =:= <<"true">>) ->
```
Comment (k32) [inline apps/emqx_authz/src/emqx_authz_api_sources.erl:449]: Nit: IS_TRUE

**Review by k32 — APPROVED** [review]

File: apps/emqx_authz/test/emqx_authz_api_sources_SUITE.erl [hunk 3]
```diff
@@ -29,17 +29,17 @@
...
                    <<"method">> => <<"get">>,
                    <<"request_timeout">> => <<"5s">>
                   }).
 -define(SOURCE2, #{<<"type">> => <<"mongodb">>,
                    <<"enable">> => true,
                    <<"mongo_type">> => <<"sharded">>,
-                   <<"servers">> => [<<"127.0.0.1:27017">>,
-                                     <<"192.168.0.1:27017">>
-                                    ],
+                   <<"servers">> => <<"127.0.0.1:27017,192.168.0.1:27017">>,
```
Comment (thalesmg) [inline apps/emqx_authz/test/emqx_authz_api_sources_SUITE.erl:42]: The type before was a list.  Does this option support both comma separated servers and a list?  Maybe it's worth to test both?

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_authz/test/emqx_authz_api_sources_SUITE.erl (hunk 3 above)
Comment (savonarola) [reply apps/emqx_authz/test/emqx_authz_api_sources_SUITE.erl:42]: These tests had been commented out, so they were a bit outdated.

Type for `servers` is binary now: 
https://github.com/emqx/emqx/blob/master/apps/emqx_connector/src/emqx_connector_mongo.erl#L253

File: apps/emqx_authz/src/emqx_authz_api_sources.erl (hunk 2 above)
Comment (savonarola) [reply apps/emqx_authz/src/emqx_authz_api_sources.erl:449]: 🆗 Will update this in one of the following authz improvements.

**Pull request merged**